### PR TITLE
feat(java): Skeleton and core components for Java encryption context exercise.

### PR DESCRIPTION
**Pull request by lizroth**
*Issue #, if available:*

*Description of changes:*
* Basic Maven setup
* Basic JUnit5 test setup
* Checkstyle and Findbugs configuration
* Automatic code formatting
* Mostly-complete Data Model
* Most (all?) final dependencies declared
* `make` target to get access to JaCoCo data from the CLI because only HTML/XML output seems to be supported at the moment
  * lol

*Incomplete/open issues:*
* Javadoc: https://github.com/aws-samples/busy-engineers-document-bucket/issues/24
* TOML config bootstrapping/plumbing (@seebees said he was taking a swing at it)
* Plumbing through to `Api` class to interact with model and AWS APIs
* Need #21 to merge to have access to the expected sort key values and prefix from TOML config

(Currently this PR is less complete than the Python version in #21)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: exercises/java/Makefile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+bootstrap:
+	wget https://d3pxv6yz143wms.cloudfront.net/11.0.5.10.1/java-11-amazon-corretto-devel-11.0.5.10-1.x86_64.rpm -O /tmp/java-corretto.rpm
```
Comment (karlw00t) [inline exercises/java/Makefile]: can't this just be yum installed?

https://docs.aws.amazon.com/corretto/latest/corretto-11-ug/amazon-linux-install.html

File: exercises/java/encryption-context-complete/Makefile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+coverage:
+	mvn clean
+	mvn test
+	cd ./target/site/jacoco/; python -m http.server &
```
Comment (karlw00t) [inline exercises/java/encryption-context-complete/Makefile]: can you say python3 here? the syntax for http server changed between 2 and 3

https://www.poftut.com/how-to-run-and-use-simple-http-server-in-python2-and-python3/

File: exercises/java/encryption-context-complete/.gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+/bin/
```
Comment (karlw00t) [inline exercises/java/encryption-context-complete/.gitignore]: why `/bin/` and not `bin/` ? I don't think there is a difference other than the inconsistency.

File: exercises/java/encryption-context-complete/pom.xml [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+    <!-- FIXME change it to the project's website -->
+    <url>http://www.example.com</url>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
+    </properties>
+
+    <dependencies>
```
Comment (karlw00t) [inline exercises/java/encryption-context-complete/pom.xml]: Do you want to add ACCP to this list?

https://aws.amazon.com/blogs/opensource/introducing-amazon-corretto-crypto-provider-accp/

File: exercises/java/encryption-context-complete/pom.xml [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+            </plugins>
+        </pluginManagement>
+        <plugins>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>exec-maven-plugin</artifactId>
+                <version>1.6.0</version>
+                <executions>
+                    <execution>
+                        <!-- TODO REPL Integration -->
```
Comment (karlw00t) [inline exercises/java/encryption-context-complete/pom.xml]: is there an issue for this todo?

File: exercises/java/encryption-context-complete/src/test/java/sfw/example/esdkworkshop/AppTest.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package sfw.example.esdkworkshop;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+/** Unit test for simple App. */
+public class AppTest {
+  /** Rigourous Test :-) */
```
Comment (karlw00t) [inline exercises/java/encryption-context-complete/src/test/java/sfw/example/esdkworkshop/AppTest.java]: This comment might not be understood well by all people doing this workshop.

File: exercises/java/encryption-context-complete/src/test/java/sfw/example/esdkworkshop/datamodel/PointerItemTest.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+public class PointerItemTest {
+
+  Map<String, String> sampleContext() {
+
+    Map<String, String> context = new HashMap<>(5);
+    context.put("region", "sp-moon-1");
+    context.put("user", "kilroy");
+    context.put("bananas", "yes");
+    context.put("fleet", "gamma");
+    context.put("oranges", "glad i didn't say bananas");
```
Comment (karlw00t) [inline exercises/java/encryption-context-complete/src/test/java/sfw/example/esdkworkshop/datamodel/PointerItemTest.java]: :+1: testing right here.

File: exercises/java/Makefile (hunk 1 above)
Comment (lizroth) [reply exercises/java/Makefile]: It wasn't in my repos when I searched for it (which surprised me), but maybe I fat fingered something.

Yep, looks like a fat finger. I'll fix.

File: exercises/java/encryption-context-complete/Makefile (hunk 2 above)
Comment (lizroth) [reply exercises/java/encryption-context-complete/Makefile]: Yes good point. I had envisioned this going away but maybe it's useful enough to hang around.

File: exercises/java/encryption-context-complete/.gitignore (hunk 3 above)
Comment (lizroth) [reply exercises/java/encryption-context-complete/.gitignore]: I think this was generated by Maven. I don't remember writing it. Anyway I will drop the trailing slash.

File: exercises/java/encryption-context-complete/pom.xml (hunk 4 above)
Comment (lizroth) [reply exercises/java/encryption-context-complete/pom.xml]: D'oh! I knew I was forgetting something. :) Absolutely!

File: exercises/java/encryption-context-complete/pom.xml (hunk 5 above)
Comment (lizroth) [reply exercises/java/encryption-context-complete/pom.xml]: Not yet, I am still deciding on this and actively talking with @seebees

File: exercises/java/encryption-context-complete/src/test/java/sfw/example/esdkworkshop/AppTest.java (hunk 6 above)
Comment (lizroth) [reply exercises/java/encryption-context-complete/src/test/java/sfw/example/esdkworkshop/AppTest.java]: This was autogenerated by Maven's "generate me a Maven project" and will be going away. :)

**Pull request merged**